code/go/0chain.net/smartcontract/dbs/event/blobber.go

```diff
@@ -136,6 +145,33 @@ func (edb *EventDb) GetBlobberCount() (int64, error) {
 	return count, res.Error
 }
 
+type AllocationQuery struct {
+	MaxChallengeCompletionTime time.Duration
+	MaxOfferDuration           time.Duration
+	ReadPriceRange             struct {
+		Min int64
+		Max int64
+	}
+	WritePriceRange struct {
+		Min int64
+		Max int64
+	}
+	Size              int
+	AllocationSize    int64
+	PreferredBlobbers []string
+}
+
+func (edb *EventDb) GetBlobbersFromParams(allocation AllocationQuery) ([]string, error) {
```

Comment:
The  should return when the number of blobbers >= numberOfBlobbers required (data shards + parity), currently it will return all.